README.md

```diff
@@ -1,3 +1,132 @@
 # repo-template
 
 **This is not an official Google product.**
+
+This template is for creating an AOD (access on demain) repository for GCP organizations, folders, and projects.
+
+After completing the installation. Follow below steps to file an AOD request.
+1. Open a PR
+  - A title and description to provide enough justification. E.g. "AOD request: debug customer issue X".
+  - Add following file at the repo root, note that any changes with below file changes will be considered AOD request and thus cannot be merged:
+    - iam.yaml - Request for IAM permissions on organization / folder / project level. To see an example of this file, refer to the example-iam.yaml file.
+  - Optionally use predefined duration labels on the PR to specify the IAM permission expiration. The duration is default to 2h if no label is used.
+
+2. PR approval
+  - Github workflows will check if the PR is an AOD request, marked by a "do_not_merge" failure workflow, and if it is a valid AOD request.
+  - Upon PR approval, the request will be handled by AOD.
+
+3. Close the PR after the request is handled successfully.
```

Comment:
```suggestion
## Make an AOD Request

**Make sure you have completed [prerequisite steps](link to the section below) to set up this repo properly.**

1. Open a PR
  - A title and description to provide enough justification. E.g. "AOD request: debug customer issue X".
  - To request IAM permissions on org/folder/project level, add an `iam.yaml` file in the repo root.
    - See an example of this file, see example in [example-iam.yaml](link to file).
    - Optionally use predefined duration labels on the PR to specify the IAM permission expiration. Otherwise a 2h default duration will be used.
  - (Later) To request on-demand `gcloud` commands, add an `gcloud.yaml` file in the repo root.

2. Checks
  - All AOD PRs will fail the check "do_not_merge" to prevent accidental merge.
  - The "Validate Request" check will report errors if there are problems in your `iam.yaml` or `gcloud.yaml`.
  
3. Getting approval
  - Ask one of the repo code owners to approve the PR
  - Upon approval, another GitHub workflow will run to grant you the IAM permissions or run the on-demand `gcloud` commands

3. Close the PR after you no longer need the access. The PR will automatically be closed after X hours (TODO).
```